web/plugin-manager/src/components/pluginBlock.tsx

```diff
@@ -0,0 +1,183 @@
+import { SetStateAction, useMemo, useState } from "react";
+import { ArrowPathIcon, TrashIcon, PlayCircleIcon } from "@heroicons/react/24/outline";
+import axiosServices from "../services/axios";
+import { AxiosResponse } from "axios";
+import { Plugin, PluginProps, PluginStatus } from "../../../shared/interfaces/IPlugin";
+import { statusHelper } from "../helpers/statusHelpers";
+
+function PluginBlock(p: PluginProps) {
+    // Callback to set error on parent
+    function sendErrorData(errorType: string, errorMessage: string) {
+        p.setErrorData(errorType, errorMessage);
+    }
+    // Data to display
+    let dataMemoized = useMemo(() => {
+        console.log("p.props useMemo", p.props);
+        return p.props;
+    }, [p.props]);
+
+    // Toggle status state
+    const [toggleStatus, setStatus] = useState(statusHelper(p.props.status));
+
+    const [playStatusClassName, setPlayStatusClassName] = useState(playStatus());
+
+    // fetch info from plugin to get fresh data on demand
+    async function fetchPluginStatus(): Promise<AxiosResponse<string>> {
+        let result: AxiosResponse<string> = {} as AxiosResponse<string>;
+        try {
+            return (result = await axiosServices.getpluginInfo(dataMemoized.id));
+        } catch (error) {
+            sendErrorData(
+                "error",
+                `Plugins infos failed to get infos from plugin name : ${dataMemoized.name} on id: ${dataMemoized.id}`
+            );
+        }
+        return result;
+    }
+
+    // Launch or stop plugin
+    async function launchAndStopPlugins() {
+        let resultPluginInfo: AxiosResponse<string>;
+        // fetch info from plugin to get fresh data
+        resultPluginInfo = await fetchPluginStatus();
+        // Update data
+        setStatus(statusHelper(resultPluginInfo.data));
+
+        let result: AxiosResponse<number>;
+        if (!toggleStatus) {
```

Comment:
why not using this.toggleStatus ? This is linked to the class question. I have the feeling that the code is unclear has you don't immeditely know where this variable come from.

Reply:
it's a state it's normal

Reply:
your answer is unclear to me. Maybe we should discuss that

Reply:
https://www.geeksforgeeks.org/differences-between-functional-components-and-class-components-in-react/

because we are in functionnal component we don't create classes and we use Hooks.

UseState is a Hook .
toggleStatus is a state defined by UseState. it's a way to store value through renders. If you don't store it in a hook your value is lost 

To access to toggleStatus. i use the variable.

Reply:
Thank you, I learned something.